### PR TITLE
fix(core): Prevent XSS via static cache dir

### PR DESCRIPTION
**Pull request by ivov**
https://linear.app/n8n/issue/SEC-61/241-persistent-cross-site-scripting

### CODE REVIEW

File: packages/core/src/NodeExecuteFunctions.ts [hunk 1]
```diff
@@ -3342,10 +3342,10 @@ function isFilePathBlocked(filePath: string): boolean {
 		return true;
 	}
 
-	//restrict access to .n8n folder and other .env config related paths
+	//restrict access to .n8n folder, ~/.cache/n8n/public, and other .env config related paths
 	if (blockFileAccessToN8nFiles) {
-		const { n8nFolder } = Container.get(InstanceSettings);
-		const restrictedPaths = [n8nFolder];
+		const { n8nFolder, staticCacheDir } = Container.get(InstanceSettings);
+		const restrictedPaths = [n8nFolder, staticCacheDir];
```
Comment (tomi) [inline packages/core/src/NodeExecuteFunctions.ts:3348]: Would it be possible to have a test for this?

File: packages/cli/BREAKING-CHANGES.md [hunk 2]
```diff
@@ -2,6 +2,16 @@
 
 This list shows all the versions which include breaking changes and how to upgrade.
 
+## 1.55.0
+
+### What changed?
+
+The `N8N_RESTRICT_FILE_ACCESS_TO` environment variable now also blocks access to n8n's static cache directory at `~/.cache/n8n/public`.
```
Comment (tomi) [inline packages/cli/BREAKING-CHANGES.md:9]: Shouldn't this be `N8N_BLOCK_FILE_ACCESS_TO_N8N_FILES`?

Comment (ivov) [reply packages/cli/BREAKING-CHANGES.md]: Good catch!

**Review by tomi — APPROVED** [review]
LGTM 🚀

**Comment by janober** [comment]
Got released with [`n8n@1.55.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.55.0)

**Pull request merged**